Clean up deprecated and unused IPCache APIs after FQDN transition to asynchronous APIs

tklauser opened the pull request:
Clean up some deprecated/unused ipcache API following #29036 (fixing issue #28930). I've noticed these could be dropped while re-reviewing that PR after merge.

See commits for details.

tklauser committed:
ipcache: fix godoc comment for (*metadata).remove

Commit 6af2b786a1dc ("ipcache/md: Make IPCache.metadata.remove()
generic") changed the method name.

Signed-off-by: Tobias Klauser <tobias@cilium.io>

tklauser committed:
ipcache: unexport (*IPCache).UpsertGeneratedIdentities

This method is deprecated. Since commit 507259fd48d6 ("fqdn: move to
asynchronous ipcache APIs") its only user is in pkg/ipcache. Unexport it
so no new code will accidentally use it.

Signed-off-by: Tobias Klauser <tobias@cilium.io>

ti-mo reviewed (approved):
I got pinged as tophat for `pkg/ip/`, could that one be assigned a codeowner?

gandro reviewed (approved):
Thanks! :recycle:

christarazi reviewed (approved):
:broom:

joestringer commented on pkg/ip/cidr.go:
```diff
@@ -83,20 +83,6 @@ func IPToNetPrefix(ip net.IP) netip.Prefix {
-// IPsToNetPrefixes returns all of the ips as a slice of netip.Prefix.
-//
-// See IPToNetPrefix() for how net.IP types are handled by this function.
-func IPsToNetPrefixes(ips []net.IP) []netip.Prefix {
```
Do you think we might still need this for some of the other remaining `net.IP` transitions? In general I figured that we'd drop all of these helpers once the migration is complete.

danehans replied:
+1

tklauser replied:
Good point. Though from a quick `grep` I don't this specific case (`[]net.IP` to `[]netip.Prefix`) to be needed, but let's keep it around for now in case it will still be needed. Will drop the last commit.

danehans reviewed (commented):
Other than https://github.com/cilium/cilium/pull/29657/files#r1417759146, /LGTM.

tklauser committed:
ipcache: remove deprecated and unused synchronous API methods

The AllocateCIDRsForIPs and ReleaseCIDRIdentitiesByID methods are
deprecated API and not used anymore since commit 507259fd48d6 ("fqdn:
move to asynchronous ipcache APIs").

Signed-off-by: Tobias Klauser <tobias@cilium.io>
